src/unit_tests/method/method_order_test.cpp

```diff
@@ -0,0 +1,181 @@
+/************************************************************************************
+*                                                                                   *
+*   Copyright (c) 2014 - 2018 Axel Menzel <info@rttr.org>                           *
+*                                                                                   *
+*   This file is part of RTTR (Run Time Type Reflection)                            *
+*   License: MIT License                                                            *
+*                                                                                   *
+*   Permission is hereby granted, free of charge, to any person obtaining           *
+*   a copy of this software and associated documentation files (the "Software"),    *
+*   to deal in the Software without restriction, including without limitation       *
+*   the rights to use, copy, modify, merge, publish, distribute, sublicense,        *
+*   and/or sell copies of the Software, and to permit persons to whom the           *
+*   Software is furnished to do so, subject to the following conditions:            *
+*                                                                                   *
+*   The above copyright notice and this permission notice shall be included in      *
+*   all copies or substantial portions of the Software.                             *
+*                                                                                   *
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      *
+*   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        *
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     *
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          *
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   *
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   *
+*   SOFTWARE.                                                                       *
+*                                                                                   *
+*************************************************************************************/
+
+#include <rttr/registration>
+#include <catch/catch.hpp>
+#include <string>
+
+using namespace rttr;
+
+struct method_order_test_base
+{
+    method_order_test_base() = default ;
+    virtual ~method_order_test_base() = default ;
+
+    std::string whoami() { return "I am a base method" ; }
+    virtual std::string vwhoami() { return "I am a virtual base method" ; }
+
+    RTTR_ENABLE()
+
+};
+
+
+// RTTR_ENABLE incompatible with clang
+#define RTTR_ENABLE_OVERRIDE(...) \
```

Comment:
What is the problem here? Why is this necessary?

Reply:
Use RTTR_ENABLE instead of RTTR_OVERRIDE and the CI will fail.

Clang in your CI fails if RTTR_ENABLE is used twice in a single file.  The reason is NOT using "override" on the virtual is an error rather than a warning.

Reply:
I have never encountered such a warning, How do you get this?
Anyway, this should be fixed globally not locally.
I could disable this clang warning, because it is in the current case wrong.